### DISCUSSION TITLE
Performance Issues on Large Dashboards after Upgrading v8 to v10

### DISCUSSION THREAD

**Discussion by macklin-fluehr** [getredash/redash#6953] [General]
I wanted to ask for some advice on how to handle a specific blocker to upgrading to v10. We have some large dashboards, which, while are slow to load in V8, do eventually load, tile by tile.

We tried upgrading to v10 and found that these dashboards just crashed. In some borderline scenarios, nothing would load and then everything loaded at once, which seems to be a different loading behavior than v8. But, in general, we would just get an error saying the page did not load, please contact your administrator.

I'm assuming there might be a way to update our docker-compose config resource count to help with this, but I'm unsure what resources in V10 might do that. I tried changing a couple with no discernible success. 

To mitigate I rolled back to v8 for the interim.

redash: docker
image: redash/redash:10.1.0.b50633

**Comment by justinclift** [comment]
That's interesting.  Do you have an idea of whether the problem is on the backend (eg retrieving data takes too long), vs the front end (eg passing too much data in the query results)?

**Reply by macklin-fluehr** [reply]
My guess is the former? While the browser is laggy (and was laggy on V8), it seems like redash just fails to get anything to the browser on v10. For example, I don't get even all the tiles to load in v10, vs in v8, I can get some of the tile outlines to load before the query results fill in. 

My assumption is that in v10, everything tries to get queried at once, which causes a timeout on everything and then rehash gives up and throws the error message.

**Reply by justinclift** [reply]
Interesting.  I remember from waaaaay back (early v10 maybe?) that the newer version of Redash seems to need more ram than the older one.

So, my first suspicion here is you *might* be running into some kind of ram limitation and things go badly into swap space.  That's without knowing anything about your setup though. :wink:

What kind of cpu and ram does your Redash have, and are you able to boost the ram massively (ie double it or something), even if just temporarily to test if things suddenly start working better?
